theano/compile/function_module.py

```diff
@@ -752,9 +753,33 @@ def checkSV(sv_ori, sv_rpl):
         return f_cpy
 
     def __call__(self, *args, **kwargs):
+        """
+        Evaluates value of a function on given arguments.
+
+        Parameters
+        ----------
+        args : list
+            List of inputs to the function. All inputs are required, even when
+            some of them are not necessary to calculate requested subset of
+            outputs.
+
+        kwargs : dict
+            TODO: other kwargs?
+            Keyword argument `output_subset` is a list of indices of the
+            function's outputs that are requested to be calculated.
```

Comment:
Just to say that if we use a different way of calling the Theano function,
we could only allow the normal input to be passed as positional arguments.

I have a slight preference to not allow kwargs parameter to use the name
output_subset. It isn't frequently used, so I don't think it would cause
problems.

I would always disallow that name in kwargs, just to prevent people from
using the bad name without partial eval and then later trying partial eval.

On Sat, Mar 19, 2016 at 6:17 AM, Jakub Sygnowski <notifications@github.com>
wrote:

> In theano/compile/function_module.py
> <https://github.com/Theano/Theano/pull/4259#discussion_r56747277>:
>
> > @@ -752,9 +753,33 @@ def checkSV(sv_ori, sv_rpl):
> >          return f_cpy
> >
> >      def __call__(self, *args, **kwargs):
> > +        """
> > +        Evaluates value of a function on given arguments.
> > +
> > +        Parameters
> > +        ----------
> > +        args : list
> > +            List of inputs to the function. All inputs are required, even when
> > +            some of them are not necessary to calculate requested subset of
> > +            outputs.
> > +
> > +        kwargs : dict
> > +            TODO: other kwargs?
> > +            Keyword argument `output_subset` is a list of indices of the
> > +            function's outputs that are requested to be calculated.
>
> It's worth noting that the issue arose at all, because to do 2) we would
> still have to pass the information about the output subset to the *call*
> anyway, and would have to do it indirectly (both all regular and keyword
> arguments are already potentially being used).
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4259/files/dbbf801533389a5d8d28608a674e97c15be46c55#r56747277>
>